tests/foreman/api/test_computeresource_libvirt.py

```diff
@@ -28,490 +28,510 @@
 
-    @pytest.mark.tier2
-    def test_negative_update_url(self):
-        """Attempt to update a compute resource with invalid url
 
-        :id: b5256090-2ceb-4976-b54e-60d60419fe50
+@pytest.mark.tier2
+@pytest.mark.parametrize('url', ['', gen_string('alpha')])
+def test_negative_create_with_url(setup, url):
+    """Attempt to create compute resources with invalid url
+
+    :id: 37e9bf39-382e-4f02-af54-d3a17e285c2a
 
-        :expectedresults: Compute resources is not updated
+    :expectedresults: Compute resources are not created
 
-        :CaseImportance: High
+    :CaseImportance: High
 
-        :CaseLevel: Component
-        """
-        compresource = entities.LibvirtComputeResource(
-            location=[self.loc], organization=[self.org], url=self.current_libvirt_url
+    :CaseLevel: Component
+
+    :parametrized: yes
+    """
+    with pytest.raises(HTTPError):
+        entities.LibvirtComputeResource(
+            location=[setup.loc], organization=[setup.org], url=url
         ).create()
-        for url in ('', gen_string('alpha')):
-            with self.subTest(url):
-                with self.assertRaises(HTTPError):
-                    compresource.url = url
-                    compresource.update(['url'])
-            self.assertNotEqual(compresource.read().url, url)
+
+
+@pytest.mark.tier2
+@pytest.mark.parametrize('new_name', invalid_values_list())
+def test_negative_update_invalid_name(setup, new_name):
+    """Attempt to update compute resource with invalid names
+
+    :id: a6554c1f-e52f-4614-9fc3-2127ced31470
+
+    :expectedresults: Compute resource is not updated
+
+    :CaseImportance: High
+
+    :CaseLevel: Component
+
+    :parametrized: yes
+    """
+    name = gen_string('alphanumeric')
+    compresource = entities.LibvirtComputeResource(
+        location=[setup.loc], name=name, organization=[setup.org], url=setup.current_libvirt_url
+    ).create()
+    with pytest.raises(HTTPError):
+        compresource.name = new_name
+        compresource.update(['name'])
+    assert compresource.read().name == name
+
+
+@pytest.mark.tier2
+def test_negative_update_same_name(setup):
+    """Attempt to update a compute resource with already existing name
+
+    :id: 4d7c5eb0-b8cb-414f-aa10-fe464a164ab4
+
+    :expectedresults: Compute resources is not updated
+
+    :CaseImportance: High
+
+    :CaseLevel: Component
+    """
+    name = gen_string('alphanumeric')
+    entities.LibvirtComputeResource(
+        location=[setup.loc], name=name, organization=[setup.org], url=setup.current_libvirt_url
+    ).create()
+    new_compresource = entities.LibvirtComputeResource(
+        location=[setup.loc], organization=[setup.org], url=setup.current_libvirt_url
+    ).create()
+    with pytest.raises(HTTPError):
+        new_compresource.name = name
+        new_compresource.update(['name'])
+    assert new_compresource.read().name != name
+
+
+@pytest.mark.tier2
+@pytest.mark.parametrize('url', ['', gen_string('alpha')])
```

Comment:
this breaks pytest collection

Reply:
This will fixi it:
`@pytest.mark.parametrize('url', **parametrized({'alpha': gen_string('alpha')}))`